battstat/acpi-linux.c

```diff
@@ -326,139 +331,144 @@ void acpi_linux_cleanup(struct acpi_info * acpiinfo)
-
-  procdir=opendir("/proc/acpi/battery/");
-  if (!procdir)
-    return FALSE;
-
-  /* Get the remaining capacity for the batteries.  Other information
-   * such as AC state and battery max capacity are read only when they
-   * change using acpi_process_event(). */
-  while ((procdirentry=readdir(procdir)))
-   {
-    if (procdirentry->d_name[0]!='.')
-     {
-      batt_state = g_strconcat("/proc/acpi/battery/",
-			       procdirentry->d_name,
-			       "/",
-			       acpiinfo->batt_state_state,
-			       NULL);
-      hash = read_file (batt_state, buf, sizeof (buf));
-      if (hash)
+    guint32 remain;
+    guint32 rate;
+    gboolean charging;
+    GHashTable *hash;
+    gchar* batt_state = NULL;
+    DIR * procdir;
+    struct dirent * procdirentry;
+    char buf[BUFSIZ];
+
+    g_assert (acpiinfo);
+
+    /*
+     * apminfo.ac_line_status must be one when on ac power
+     * apminfo.battery_status must be 0 for high, 1 for low, 2 for critical, 3 for charging
+     * apminfo.battery_percentage must contain batter charge percentage
+     * apminfo.battery_flags & 0x8 must be nonzero when charging
+     */
+
+    g_assert (apminfo);
+
+    charging = FALSE;
+    remain = 0;
+    rate = 0;
+
+    procdir=opendir ("/proc/acpi/battery/");
+    if (!procdir)
+        return FALSE;
+
+    /* Get the remaining capacity for the batteries.  Other information
+     * such as AC state and battery max capacity are read only when they
+     * change using acpi_process_event (). */
+    while ((procdirentry=readdir (procdir)))
        {
-        const char *s;
-        if (!charging)
-         {
-          s = read_string (hash, acpiinfo->charging_state);
-          charging = s ? (strcmp (s, "charging") == 0) : 0;
-         }
-        remain += read_long (hash, "remaining capacity");
-	rate += read_long (hash, "present rate");
-        g_hash_table_destroy (hash);
+          if (procdirentry->d_name[0]!='.')
+              {
+                  batt_state = g_strconcat ("/proc/acpi/battery/",
+                                            procdirentry->d_name,
+                                            "/",
+                                            acpiinfo->batt_state_state,
+                                            NULL);
+                  hash = read_file (batt_state, buf, sizeof (buf));
+                  if (hash)
+                      {
+                          const char *s;
+                          if (!charging)
+                              {
+                                  s = read_string (hash, acpiinfo->charging_state);
+                                  charging = s ? (strcmp (s, "charging") == 0) : 0;
+                              }
+                          remain += read_long (hash, "remaining capacity");
+                          rate += read_long (hash, "present rate");
+                          g_hash_table_destroy (hash);
+                      }
+                  g_free (batt_state);
+              }
```

Comment:
Use Allman or K&R [indent style](https://en.wikipedia.org/wiki/Indentation_style#Styles).

Reply:
now there is more hegemony in the coding style, and it can be read better.